code/game/objects/items/weapons/grenades/ghettobomb.dm

```diff
@@ -82,9 +105,29 @@
 			spawn(det_time)
 				prime()
 
+
+/obj/item/weapon/grenade/iedcasing/proc/AddShrapnel(var/obj/item/I, mob/user as mob)
+
+	if(assembled == 2)
+		if((current_shrapnel + I.shrapnel_size)<= max_shrapnel )
+			if(I.shrapnel_amount > 0|| I.w_class == W_CLASS_TINY)
+				shrapnel_list.Add(I)
+				current_shrapnel += I.shrapnel_size
+				if(user && user.drop_item(I, src))
+					to_chat(user, "<span  class='notice'>You add \the [I] to the improvised explosive.</span>")
+					playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 25, 1)
+				else
+					I.forceMove(src)
+
+	else
+		if(user)
+			to_chat(user, "<span  class='notice'>There is no room for \the [I] in the improvised explosive!.</span>")
```

Comment:
Uh, this would be shown if the assembly state isn't 2, not when there's no more room for shrapnel. I think you want to indent this some more.